src/connectors/autocomplete/connectAutocomplete.ts

```diff
@@ -103,35 +103,44 @@ search.addWidgets([
       $$type: 'ais.autocomplete',
 
       init(initOptions) {
-        const { helper, renderState, instantSearchInstance } = initOptions;
+        const { helper, instantSearchInstance } = initOptions;
         connectorState.refine = (query: string) => {
           helper.setQuery(query).search();
         };
 
         renderFn(
           {
-            ...this.getWidgetRenderState!(renderState, initOptions)
-              .autocomplete!,
+            ...(this.getWidgetRenderState!(
+              initOptions
+            ) as AutocompleteWidgetRenderState),
```

Comment:
No way to type the function return type instead of casting it here ? (casting this way is not type safe)

Reply:
We can probably make `getWidgetRenderState` generic?

Reply:
I tried to find solutions to make the whole widget generic based on the type passed to the connector, but I couldn't really make it work. Just having gwrs generic will be a good stop-gap